lib/HLSL/DxilCondenseResources.cpp

```diff
@@ -1821,18 +1821,17 @@ bool UpdateStructTypeForLegacyLayout(DxilResourceBase &Res,
       Function *NF = hlslOP->GetOpFunc(hlsl::OP::OpCode::CreateHandleForLib, UpdatedST);
 
       // Replace old GV.
-      for (auto UserIt = Symbol->user_begin(); UserIt != Symbol->user_end();) {
+      for (auto UserIt = Symbol->user_begin(), userEnd = Symbol->user_end(); UserIt != userEnd;) {
```

Comment:
Is there a reason for this transformation? I feel like the loop was cleaner to read before this change.

Reply:
It was an attempt to track down a different bug, but the user can get deleted in this loop and that is a condition where this form of the loop has been required before. Even though I couldn't get a crash or a memory misuse complaint here, this seems the safer route.